code/glTF2Exporter.cpp

```diff
@@ -413,48 +414,62 @@ void glTF2Exporter::ExportMaterials()
-        }
-
-        GetMatTex(mat, m->pbrMetallicRoughness.metallicRoughnessTexture, AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_METALLICROUGHNESS_TEXTURE);
-
-        if (GetMatColor(mat, m->pbrMetallicRoughness.baseColorFactor, AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_BASE_COLOR_FACTOR) != AI_SUCCESS) {
-            // if baseColorFactor wasn't defined, then the source is likely not a metallic roughness material.
-            //a fallback to any diffuse color should be used instead
-            GetMatColor(mat, m->pbrMetallicRoughness.baseColorFactor, AI_MATKEY_COLOR_DIFFUSE);
-        }
-
-        if (mat->Get(AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_METALLIC_FACTOR, m->pbrMetallicRoughness.metallicFactor) != AI_SUCCESS) {
-            //if metallicFactor wasn't defined, then the source is likely not a PBR file, and the metallicFactor should be 0
-            m->pbrMetallicRoughness.metallicFactor = 0;
-        }
-
-        // get roughness if source is gltf2 file
-        if (mat->Get(AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_ROUGHNESS_FACTOR, m->pbrMetallicRoughness.roughnessFactor) != AI_SUCCESS) {
-            // otherwise, try to derive and convert from specular + shininess values
-            aiColor4D specularColor;
-            ai_real shininess;
-
-            if (
-                mat->Get(AI_MATKEY_COLOR_SPECULAR, specularColor) == AI_SUCCESS &&
-                mat->Get(AI_MATKEY_SHININESS, shininess) == AI_SUCCESS
-            ) {
-                // convert specular color to luminance
-                float specularIntensity = specularColor[0] * 0.2125f + specularColor[1] * 0.7154f + specularColor[2] * 0.0721f;
-                //normalize shininess (assuming max is 1000) with an inverse exponentional curve
-                float normalizedShininess = std::sqrt(shininess / 1000);
-
-                //clamp the shininess value between 0 and 1
-                normalizedShininess = std::min(std::max(normalizedShininess, 0.0f), 1.0f);
-                // low specular intensity values should produce a rough material even if shininess is high.
-                normalizedShininess = normalizedShininess * specularIntensity;
-
-                m->pbrMetallicRoughness.roughnessFactor = 1 - normalizedShininess;
+        
+        // The value is true if the material is from common material source or the export option is true
+        bool hasCommonSource = false;
+        mat->Get(AI_MATKEY_GLTF_COMMON, hasCommonSource);
+        bool exportAsCommonMaterial = hasCommonSource || (mProperties->HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_COMMON) && mProperties->GetPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_COMMON));
+        // The value is false if the export as a common material is active and the export option is false
+        bool exportAsPBRMaterial = !(exportAsCommonMaterial && mProperties->HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR) && !mProperties->GetPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR));
```

Comment:
Isn't there an issue if `HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR)` is false? Shouldn't it be true for the `GetProperty` to be safe or is there a default?

Reply:
The default value for AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR  is True (I added the description in config.h.in). If I can't guarantee that only the KHR_materials_common is needed by explicitly deactivating the pbr export and activating the extension, the pbr is exported.